core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/internal/UserStoreMgtDataHolder.java

```diff
@@ -47,4 +52,40 @@ public CryptoService getCryptoService() {
         return cryptoService;
     }
 
+    /**
+     * Set each HashProviderFactory instance in a Map.
+     *
+     * @param hashProviderFactory Instance of HashProviderFactory.
+     */
+    public void setHashProviderFactory(HashProviderFactory hashProviderFactory) {
+
+        if (hashProviderFactoryMap == null) {
+            hashProviderFactoryMap = new HashMap<>();
+        }
+        hashProviderFactoryMap.put(hashProviderFactory.getType(), hashProviderFactory);
+    }
+
+    /**
+     * Get the HashProviderFactory instance from Map.
```

Comment:
`Get the HashProviderFactory from HashProviderFactory collection.`

Reply:
Fix other places accordingly

Reply:
Fixed